source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/network/connection.h"
+
+#include "common/common/empty_string.h"
+#include "common/common/logger.h"
+#include "common/network/filter_manager_impl.h"
+#include "common/stream_info/stream_info_impl.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_connection.h"
+
+namespace Envoy {
+namespace Quic {
+
+// Act as a Network::Connection to HCM and a FilterManager to FilterFactoryCb.
+class QuicFilterManagerConnectionImpl : public Network::FilterManagerConnection,
+                                        protected Logger::Loggable<Logger::Id::connection> {
+public:
+  QuicFilterManagerConnectionImpl(std::unique_ptr<EnvoyQuicConnection> connection,
+                                  Event::Dispatcher& dispatcher);
+
+  // Network::FilterManager
+  // Overridden to delegate calls to filter_manager_.
+  void addWriteFilter(Network::WriteFilterSharedPtr filter) override;
+  void addFilter(Network::FilterSharedPtr filter) override;
+  void addReadFilter(Network::ReadFilterSharedPtr filter) override;
+  bool initializeReadFilters() override;
+
+  // Network::Connection
+  void addConnectionCallbacks(Network::ConnectionCallbacks& cb) override;
+  void addBytesSentCallback(Network::Connection::BytesSentCb /*cb*/) override;
+  void enableHalfClose(bool enabled) override;
+  void close(Network::ConnectionCloseType type) override;
+  Event::Dispatcher& dispatcher() override { return dispatcher_; }
+  uint64_t id() const override {
+    // QUIC connection id can be 18 types. It's easier to use hash value instead
+    // of trying to map it into a 64-bit space.
+    return quic_connection_->connection_id().Hash();
```

Comment:
Is this cached or computed each time? This might get called a lot.

Reply:
Good point. It's computed each time using boringSSL siphash: https://cs.chromium.org/chromium/src/net/third_party/quiche/src/quic/core/quic_connection_id.cc?rcl=e3f2f7bd0dddec518b59fed97a3a6d48cbbd1f59&l=43

I can latch it in this class for return. This will make life time management easier as well.